packages/backend/src/analytics/LightdashAnalytics.ts

```diff
@@ -1564,4 +1561,33 @@ export class LightdashAnalytics extends Analytics {
             context: { ...this.lightdashContext }, // NOTE: spread because rudderstack manipulates arg
         });
     }
+
+    /**
+     * Track events from an account. This lets us centralize common properties that are found in req.account.
+     */
+    trackAccount<T extends BaseTrack>(
+        account: Account,
+        basePayload: TypedEvent | UntypedEvent<T>,
+    ) {
+        const payload = {
+            ...basePayload,
+        };
+
+        if (!payload.properties) {
+            payload.properties = {};
+        }
+
+        if (account.authentication.type === 'session') {
+            payload.userId = account.user.id;
+        } else {
+            payload.anonymousId = 'embed';
+            payload.properties.externalId = account.user.id;
+        }
+
+        // TODO: Could these be top-level fields rather than properties?
```

Comment:
I agree, but we need to confirm first how this is used and how easy will be to migrate on analtyics